frontend/views/containers/proposals/RemoveMember.vue

```diff
@@ -1,58 +1,129 @@
 <template lang="pug">
 proposal-template(
+  ref='proposal'
+  variant='removeMember'
   :title='L("Remove Member")'
-  :rule='{ value: 7, total: 10 }'
+  :rule='rule'
   :maxSteps='config.steps.length'
   :currentStep.sync='ephemeral.currentStep'
   @submit='submit'
 )
   .c-step(v-if='ephemeral.currentStep === 0' key='0')
-    avatar.c-avatar(:src='member.picture' size='lg')
-    i18n.is-title-4(tag='p' :args='{ name: member.displayName || member.name }') Remove {name} from your group
+    avatar.c-avatar(:src='memberGlobalProfile.picture' size='lg')
+    p.is-title-4.c-descr(data-test='description')
+      i18n(:args='{ name: userDisplayName(username) }' v-if='groupShouldPropose') Remove {name} from the group
+      i18n(:args='{ name: userDisplayName(username) }' v-else) Are you sure you want to remove {name} from the group?
+
+  banner-scoped(ref='formMsg' data-test='proposalError')
 </template>
 
 <script>
+import { mapState, mapGetters } from 'vuex'
+import sbp from '~/shared/sbp.js'
+import { CLOSE_MODAL } from '@utils/events.js'
+import L from '@view-utils/translations.js'
 import Avatar from '@components/Avatar.vue'
+import { PROPOSAL_REMOVE_MEMBER } from '@model/contracts/voting/constants.js'
+import BannerScoped from '@components/banners/BannerScoped.vue'
 import ProposalTemplate from './ProposalTemplate.vue'
 
 export default {
   name: 'RemoveMember',
   components: {
     Avatar,
+    BannerScoped,
     ProposalTemplate
   },
+  props: {
+    username: String
+  },
   data () {
     return {
       ephemeral: {
         currentStep: 0
       },
       config: {
         steps: [
-          'AddMemberRule'
+          'RemoveMember'
         ]
       }
     }
   },
   created () {
-    console.log('TODO - Get username info')
+    if (!this.username) {
+      console.warn('Missing username to display RemoveMember modal')
+      sbp('okTurtles.events/emit', CLOSE_MODAL)
+    }
   },
   computed: {
-    member () {
-      // TODO - connect to real store.
-      return {
-        id: 1000,
-        displayName: '[Attila the Hun]',
-        picture: '/assets/images/default-avatar.png'
-      }
+    ...mapState([
+      'currentGroupId'
+    ]),
+    ...mapGetters([
+      'currentGroupState',
+      'globalProfile',
+      'groupProfiles',
+      'groupSettings',
+      'groupShouldPropose',
+      'groupMembersCount',
+      'ourUsername',
+      'userDisplayName'
+    ]),
+    memberGlobalProfile () {
+      return this.globalProfile(this.username) || {}
+    },
+    rule () {
+      const { threshold } = this.groupSettings.proposals['remove-member'].ruleSettings.threshold
```

Comment:
Instead of `'remove-member'`, we must use the constant `PROPOSAL_REMOVE_MEMBER`.

Also, can you please check here and elsewhere (below and it seems even in other files), what's gong on with the value of `.ruleSettings`?

The idea was originally to make it so that each proposal could have a different voting rule applied to it, and that this could be changed by groups.

Which rule is applied to which proposal is stored in `getters.groupSettings.proposals.[<RULE_TYPE>].rule`.

And therefore to get the threshold this line actually needs to be:

```js
const ruleType = this.groupSettings.proposals.[PROPOSAL_REMOVE_MEMBER].rule
const { threshold } = this.groupSettings.proposals[PROPOSAL_REMOVE_MEMBER].ruleSettings[ruleType]
```

I've noticed that this is an error in may other files. In this PR, can you also do some boyscoutting by searching the project for `ruleSettings.threshold` ?

This error also happens in:

- AddMembers.vue
- Mincome.vue

Reply:
> I've noticed that this is an error in may other files. In this PR, can you also do some boyscoutting by searching the project for ruleSettings.threshold ?

Done.

Regarding the `RULE_DISAGREEMENT` needed modifications. This also involves design work from @mmbotelho. So I'd rather wait for her part (#409) and address it there, instead of creating another "related-duplicated" issue. Okay?

Reply:
It's OK, I'll create the issue just to make sure we don't forget, since #409 doesn't mention this. Adding this to my TODO list. Thanks for pushing the recent fixes, about to approve!

Reply:
Please leave this thread unresolved so that it can be linked to from issue #853